src/python/pants/backend/codegen/tasks/protobuf_gen.py

```diff
@@ -103,55 +86,44 @@ def resolve_deps(self, deps_list, key):
                                   .format(message=e, key=key))
     return deps
 
+  def invalidate_for_files(self):
+    return [self.protobuf_binary]
+
   @property
   def javadeps(self):
     return self.resolve_deps(self.get_options().javadeps, 'javadeps')
```

Comment:
nit: I think it might have been important to make this a property when javadeps was stored as a local attribute and the resolve logic was more extensive  

```
self._javadeps = ...
```

but now the code is so concise, maybe we should just inline it into synthetic_target_extra_dependencies (unless there is a reason you want to unit test it)